### PR TITLE
fix(MenuGroup): Add option to use React component and Node in groupMenu

### PR DESCRIPTION
**Pull request by karelhala**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/8209

**Additional issues**:
Not sure if it fails or not, locally after downloading master, installing and running build/tests it fails.

### CODE REVIEW

**Comment by nicolethoen** [comment]
Looks like MenuDemo.tsx has a type issue on line 373
`Type '({ className }: { className: any; }) => Element' is not assignable to type 'string | Node | ReactElement<any, string | JSXElementConstructor<any>>'.`

**Comment by karelhala** [comment]
@nicolethoen ah yeah, typescript bit me again. I've extracted the component to specific class so TS won't complain.

**Comment by nicolethoen** [comment]
@karelhala do you have more code you intended to push to this PR? just making sure in case you thought you'd already updated it.

**Comment by karelhala** [comment]
@nicolethoen I had a hard time figuring out the types, but looks like everything is running!

**Review by nicolethoen — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Comment by karelhala** [comment]
@nicolethoen what's the process of mergning tickets? I have the green button, but I don't want to mess with your release cycle.

File: packages/react-core/src/components/Menu/MenuGroup.tsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import * as React from 'react';
...
 
-export interface MenuGroupProps extends React.HTMLProps<HTMLElement> {
+export interface MenuGroupProps extends Omit<React.HTMLProps<HTMLElement>, 'label'> {
   /** Items within group */
   children?: React.ReactNode;
   /** Additional classes added to the MenuGroup */
   className?: string;
   /** Group label */
-  label?: string;
+  label?: string | React.ReactNode | React.FC;
```
Comment (tlabaj) [inline packages/react-core/src/components/Menu/MenuGroup.tsx:11]: You can leave string out since `React.ReactNode` includes strings.

**Review by tlabaj — CHANGES_REQUESTED** [review]
@karelhala Sorry for the delay we just wrapped up code freeze a couple of days ago.  I have small nit.
I also have a question but adding `React.FC` to the type.  Can we be more specific with the function definition?

File: packages/react-core/src/components/Menu/MenuGroup.tsx (hunk 1 above)
Comment (karelhala) [reply packages/react-core/src/components/Menu/MenuGroup.tsx]: Sure thing!

[the remark this answers is not in the document]
Comment (karelhala) [reply packages/react-core/src/components/Menu/MenuGroup.tsx]: Sure thing!

**Comment by karelhala** [comment]
Updated props, thank you @tlabaj for review! Sorry for such a late reply, but I missed the commit review notification.

File: packages/react-core/src/components/Menu/MenuGroup.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (karelhala) [reply packages/react-core/src/components/Menu/MenuGroup.tsx]: Tried it, but that fails the build. Any reason for not using type `React.FC`?

**Review by tlabaj — APPROVED** [review]
lgtm

**Pull request merged**